.gitignore

```diff
@@ -0,0 +1,2 @@
+.idea/
+target/
```

Comment:
Why .iml file has been removed  from gitignore?

Reply:
it was not generated with maven. If we continue with maven, should I still keep it?

Reply:
Apparently, I just realized I did not pull the latest `main` to my local `main`, so I branched out `maven_task` from the outdated version of local `main`. Hence, the .gitignore file was generated from scratch and was missing `*.iml` entry. 

I just merged the latest main into this branch, so it should be ready now. 

Please, re-review.